modules/apps/portal-vulcan/portal-vulcan-test/src/testIntegration/java/com/liferay/portal/vulcan/internal/jaxrs/context/provider/test/AcceptLanguageContextProviderTest.java

```diff
@@ -151,6 +152,14 @@ private void _testCreateContext(
 
 		// One partial locale
 
+		acceptLanguage = _contextProvider.createContext(
+			new MockMessage(
+				new AcceptLanguageMockHttpServletRequest(
+					method, user, LocaleUtil.ENGLISH)));
+
+		Assert.assertEquals(
```

Comment:
I believe this test case is invalid. The algorithm used to get the locale is based on https://www.rfc-editor.org/rfc/rfc4647 by using `Locale.filter`. That function filters the available locales based on the user language priority list sent in the `Accept-Language` header. The resulting filtered list keeps the same order of the available locales unless weights are specified in the priority list (which is not your case). This means:
1.  `Locale.filter(en, [es, en, en-US])` = `[en, en-US]`.
2.  `Locale.filter(en, [es, en-US, en-US])` = `[en-US, en]`.

We have no way to know which locale is more suitable to your needs as both `en` and `en-US` match the priority list sent by the user (`en`). That's why we default to the first match, in your case `en-GB`.

The right way to get the desired `en` locale instead would be to move it up in the list of available locales above `en-UK` and `en-US`.